stubs/stripe/stripe/api_resources/invoice.pyi

```diff
@@ -11,12 +12,12 @@ from stripe.api_resources.abstract import (
 
 class Invoice(CreateableAPIResource, DeletableAPIResource, ListableAPIResource, UpdateableAPIResource):
     OBJECT_NAME: str
-    def finalize_invoice(self, idempotency_key: Any | None = ..., **params): ...
-    def mark_uncollectible(self, idempotency_key: Any | None = ..., **params): ...
-    def pay(self, idempotency_key: Any | None = ..., **params): ...
-    def send_invoice(self, idempotency_key: Any | None = ..., **params): ...
-    def void_invoice(self, idempotency_key: Any | None = ..., **params): ...
+    def finalize_invoice(self: Self, idempotency_key: str | None = ..., **params) -> Self: ...
+    def mark_uncollectible(self: Self, idempotency_key: str | None = ..., **params) -> Self: ...
+    def pay(self: Self, idempotency_key: str | None = ..., **params) -> Self: ...
+    def send_invoice(self: Self, idempotency_key: str | None = ..., **params) -> Self: ...
+    def void_invoice(self: Self, idempotency_key: str | None = ..., **params) -> Self: ...
     @classmethod
     def upcoming(
         cls, api_key: Any | None = ..., stripe_version: Any | None = ..., stripe_account: Any | None = ..., **params
-    ): ...
+    ) -> Invoice: ...
```

Comment:
And here (also, are you sure this is `Invoice` and not `Self`? Seems a bit suspect given some of the other annotations 🤔)

Reply:
yeah I think it should be Self, went with Invoice since I was being lazy and I think most people probably aren't subclassing these

Reply:
```python
>>> type(stripe.Invoice.upcoming(subscription="sub_1JrD7ZGBs2RJTknIQSCKG7Xk"))
<class 'stripe.api_resources.invoice.Invoice'>
```

Reply:
actually, looking at the code I think this will always be an Invoice since it hits the `/upcoming` invoices endpoint and then creates a Stripe object based off that (which will always be an invoice)

https://github.com/stripe/stripe-python/blob/97ee54f5e54b4af1d474de3059b26a6b92dca53d/stripe/api_resources/invoice.py#L66-L77